### PR TITLE
Fix Task::blocking(…).detach()

### PR DESCRIPTION
**Pull request by Byron**
As opposed to Task::spawn(…).detach(), blocking tasks don't complete
unless their receiving future is polled.

Previously the future would just be dropped, which caused the blocking
future not to be evaluated.

Now the future is spawned and detached to assure it gets completed
eventually and automatically.

Please note that this is my 'brute force' workaround that can also be applied manually by doing `smol::Task::spawn(smol::Task::blocking(…)).detach()`, but at least it fixes otherwise surprising and probably wrong beaviour.

The `task::blocking_detach(…)` test breaks unless the fix is applied.

### CODE REVIEW

File: src/async_io.rs [hunk 1]
```diff
@@ -19,8 +19,8 @@ use std::{
     path::Path,
 };
 
-use futures_util::future;
 use futures_io::{AsyncRead, AsyncWrite};
+use futures_util::future;
```
Comment (Byron) [inline src/async_io.rs:23]: This change was done automatically by `cargo fmt`.

**Comment by stjepang** [comment]
Thanks for the PR! I've reverted some commits and removed the use of the `blocking` crate for now.

I will re-introduce it in smol 0.2 or 1.0 soon, though.
